src/Compilers/CSharp/Portable/BoundTree/DecisionTree.cs

```diff
@@ -511,20 +511,20 @@ private DecisionTree AddByValue(DecisionTree.Guarded guarded, BoundExpression va
                 guarded.Default = new DecisionTree.ByValue(guarded.Expression, guarded.Type, null);
             }
 
-            return AddByValue(guarded.Default, value, makeDecision, hasErrors);
+            return AddByValue(guarded.Default, value, makeDecision);
         }
 
-        private DecisionTree AddByValue(DecisionTree.ByValue byValue, BoundExpression value, DecisionMaker makeDecision, bool hasErrors)
+        private DecisionTree AddByValue(DecisionTree.ByValue byValue, BoundExpression value, DecisionMaker makeDecision)
         {
             Debug.Assert(value.Type == byValue.Type);
             if (byValue.Default != null)
             {
-                return AddByValue(byValue.Default, value, makeDecision, hasErrors);
+                return AddByValue(byValue.Default, value, makeDecision);
             }
 
             // For error recovery, to avoid "unreachable code" diagnostics when there is a bad case
             // label, we use the case label itself as the value key.
-            object valueKey = hasErrors ? (object)value : value.ConstantValue.Value;
+            object valueKey = value.ConstantValue?.Value ?? value;
```

Comment:
This line is the only real change in this PR. Once this line changed, `hasErrors` was unused and could be removed.